globus_automate_client/cli/flows.py

```diff
@@ -57,6 +60,30 @@ class ActionStatus(str, Enum):
     inactive = "INACTIVE"
 
 
+class FlowInputFormat(str, Enum):
+    json = "json"
+    yaml = "yaml"
+
+
+def _process_definition(definition: str, input_format: str) -> Mapping[str, Any]:
+    """
+    Turn input strings into dicts per input format type (FlowInputFormat)
+    """
+    flow_dict = None
+    if input_format == FlowInputFormat.json:
+        try:
+            flow_dict = json.loads(definition)
+        except json.JSONDecodeError as e:
+            raise typer.BadParameter(f"Invalid JSON: {e}")
+    elif input_format == FlowInputFormat.yaml:
+        try:
+            flow_dict = yaml.safe_load(definition)
+        except Exception as e:
+            raise typer.BadParameter(f"Invalid YAML: {e}")
+
+    return flow_dict
```

Comment:
I think it may be easier to take an input string and just try loading it as JSON first - if it triggers a `JSONDecodeError`, we can try loading it as YAML - and if that still triggers an `Exception`, we can raise a `typer.BadParameter` exception. This way, we remove the burden of needing to indicate the input format from the user.

Reply:
@urimandujano Yeah, just to be clear, @jakeglobus  and I talked about this. We might end up going that way in due time, but for first cut, we decided to make it a user-set option in this first pass.

Reply:
Ah I didn't realize there'd been a discussion about the approach. Then that sounds good and the implementation looks like a good one!